src/hotspot/share/runtime/globals.hpp

```diff
@@ -2489,6 +2489,15 @@ define_pd_global(uint64_t,MaxRAM,                    1ULL*G);
           "Special mode: JVM reads a class list, loads classes, builds "    \
           "shared spaces, and dumps the shared spaces to a file to be "     \
           "used in future JVM runs")                                        \
+  /* Google: Perform parallel class loading, linking and other          */  \
+  /*         parallel-capable preprocessing operations in different     */  \
+  /*         Java threads to speed up CDS archive creation. The number  */  \
+  /*         of Java threads is determined by DumpWithParallelism       */  \
+  /*         value. If DumpWithParallelism is 1, all CDS archiving      */  \
+  /*         operations are done using one thread.                      */  \
+  product(int, DumpWithParallelism, 4,                                      \
```

Comment:
Currently, loading and linking can be done in parallel during the archive dump process. I had a plan to move more operations to the parallel phase. For example, the explicit class initialization for pre-init support can be moved to parallel phase. We have an Google internal bug for tracking that but I haven't had time to implement it yet.

Let me know if you have any suggestion for a more specific option name, so we can improve this.

Reply:
Aha.... Maybe something like "ParallelPreloadClasses" as all parallelable subphases(loading, linking and pre-initialization in the future) can be summarized as "class preloading" process. Just IMHO, please feel free to ignore this if you think "DumpWithParallelism" exactly expressed what you want to.

Reply:
The value of the option (e.g. -XX:DumpWithParallelism=8) specifies the number of threads used for parallel operations at CDS dump time. -XX:ParallelPreloadClasses=<value> would cause confusion. Maybe -XX:CDSPreProcessorWithParallelism=<value>? 

Ok, we can bring up this during  https://bugs.openjdk.java.net/browse/JDK-8278632 PR review and check in the patch in the incubator for now.

A note on the current default value: 

4 is chosen for our build environment due to the unique configuration. For any other systems, higher Parallelism probably should be used instead.

Reply:
Yes, I forget to comment that num of thread can be automatically tuned by current machine configuration, this can be a follow-up enhancement in upstream after JDK-8278632 merged.